### PR TITLE
Safeloader module split

### PR DESCRIPTION
**Pull request by clebergnu**
This splits the somewhat large (and more complex than necessary) safeloader.py file into a number of smaller, more "self contained" modules.

This should be the last refactor before the fix for #4625

### CODE REVIEW

**Review by richtja — APPROVED** [review]
LGTM, thanks

File: avocado/core/safeloader/__init__.py [hunk 1]
```diff
@@ -19,271 +19,16 @@
 
 import ast
 import collections
```
Comment (beraldoleal) [inline avocado/core/safeloader/__init__.py:21]: This split is more than welcome. And I understand the reasons to keep the code inside `__init__.py`, however I would like to suggest to move this to a different module (i.e: `safeloader/core.py`) and keep here only initialization code, and/or imports.

File: avocado/core/safeloader/module.py [hunk 2]
```diff
@@ -60,6 +42,24 @@ def __init__(self, path, module='avocado', klass='Test'):
         with open(self.path) as source_file:
             self.mod = ast.parse(source_file.read(), self.path)
 
+    @staticmethod
```
Comment (beraldoleal) [inline avocado/core/safeloader/module.py:45]: Nice!

**Review by beraldoleal — COMMENTED** [review]
Hi @clebergnu this LGTM, just one nitpicking comment.

**Review by willianrampazzo — APPROVED** [review]
With or without https://github.com/avocado-framework/avocado/pull/4649/files#r646542096, LGTM!

**Comment by clebergnu** [comment]
Closing in favor of https://github.com/avocado-framework/avocado/pull/4651

**Pull request closed without merging**